src/transaction/log_manager.c

```diff
@@ -4738,9 +4761,56 @@ log_append_commit_log_with_lock (THREAD_ENTRY * thread_p, LOG_TDES * tdes, LOG_L
 static void
 log_append_abort_log (THREAD_ENTRY * thread_p, LOG_TDES * tdes, LOG_LSA * abort_lsa)
 {
+  if (prm_get_bool_value (PRM_ID_SUPPLEMENTAL_LOG) == true)
+    {
+      log_append_supplement_user (thread_p, tdes, LOG_PRIOR_LSA_WITHOUT_LOCK);
+    }
   log_append_donetime_internal (thread_p, tdes, abort_lsa, LOG_ABORT, LOG_PRIOR_LSA_WITHOUT_LOCK);
 }
 
+/*
+ * log_append_supplement_user - append supplement user log record 
+ *
+ * return: nothing
+ *
+ *   tdes(in/out): State structure of transaction containing transaction user.
+ *   
+ */
+void
+log_append_supplement_user (THREAD_ENTRY * thread_p, LOG_TDES * tdes, int with_lock)
+{
+  LOG_REC_SUPPLEMENT_TRAN_USER *spplmnt_usr;
+  LOG_PRIOR_NODE *node;
+
+  assert (prm_get_bool_value (PRM_ID_SUPPLEMENTAL_LOG) == true);
+
+  node = prior_lsa_alloc_and_copy_data (thread_p, LOG_SUPPLEMENT_TRAN_USER, RV_NOT_DEFINED, NULL, 0, NULL, 0, NULL);
+  if (node == NULL)
+    {
+      return;
+    }
+
+  spplmnt_usr = (LOG_REC_SUPPLEMENT_TRAN_USER *) node->data_header;
+  memcpy (spplmnt_usr->user_name, tdes->client.get_db_user (), DB_MAX_USER_LENGTH);
+  if (with_lock == LOG_PRIOR_LSA_WITH_LOCK)
+    {
+      (void) prior_lsa_next_record_with_lock (thread_p, node, tdes);
+    }
+  else
+    {
+      (void) prior_lsa_next_record (thread_p, node, tdes);
+    }
+}
+
+static void
+log_append_supplement_user_at_start (THREAD_ENTRY * thread_p, LOG_TDES * tdes, int with_lock)
```

Comment:
I can't find the use case of 'with_lock != LOG_PRIOR_LSA_WITHOUT_LOCK'. So, is it necessary to use this parameter?